src/veh_interact.cpp

```diff
@@ -957,6 +944,16 @@ void veh_interact::do_refill()
         break; // no reason, all is well
     }
 
+    std::vector<vehicle_part *> ptanks;
+    auto opts = g->u.items_with( []( const item &e ) { return e.is_ammo(); } );
```

Comment:
This calls `is_ammo`, the other loop in `cant_do` calls `count_by_charges`, should be the same in both places.

In fact, why not move this loop into a function that returns the `ptanks` vector?